invenio_stats/aggregations.py

```diff
@@ -191,10 +187,35 @@ def get_indices_with_alias(self, alias_name):
 
     def run(self):
         """Calculate statistics aggregations."""
-        self.new_bookmark = (datetime.datetime.utcnow() -
-                             datetime.timedelta(hours=1)).\
-            strftime(self.doc_id_suffix)
-        bulk(self.client,
-             self.agg_iter(),
-             stats_only=True)
-        self.set_bookmark()
+        td = datetime.datetime.utcnow() - self.get_bookmark()
+        if td.days > self.batch_size:
+            lower_limit = self.get_bookmark()
+            upper_limit = lower_limit + datetime.timedelta(
+                days=self.batch_size)
+            indices = self.get_indices_with_alias('events-stats-{}'.
+                                                  format(self.event))
+            indices.sort()
+            last_index = indices[-1]
+            last_index_date = datetime.datetime.\
```

Comment:
It seems to me that this part could be simplified by just using *today* as the last_index_date.